File: app/src/main/java/com/example/app/ui/list/ViewEventListener.java
package com.example.app.ui.list;

import android.util.Log;

import com.example.app.model.JobPosting;

public class ViewEventListener {
    private static final String TAG = ViewEventListener.class.getSimpleName();
    private InteractionListener interactionListener;

    public ViewEventListener(InteractionListener listener) {
        this.interactionListener = listener;
    }

    public void onClick(JobPosting job) {
        Log.d(TAG, "onClick: " + job.getCompany());
        interactionListener.onListItemClick(job.getId());
    }
}

File: app/src/main/java/com/example/app/ui/list/ViewEventListener.kt
package com.example.app.ui.list

import android.util.Log

import com.example.app.model.JobPosting

class ViewEventListener(private val interactionListener: InteractionListener) {

    fun onClick(job: JobPosting) {
        Log.d(TAG, "onClick: " + job.company!!)
        interactionListener.onListItemClick(job.id)
    }

    companion object {
        private val TAG = ViewEventListener::class.java.simpleName
    }
}
